### PR TITLE
Add metrics plan; Define event methods and objects

### LINKED ISSUE #130: iOS Telemetry Metrics Plan
We need to collect event telemetry to tell us the how often and in what ways users are interacting with the iOS app. To this end I (Leif) will prepare a data collection plan that will include the following:

1. Which events will be collected, and when, according the latest invision prototype.
2. Each event's taxonomy: their `category`, `method`, `object`, `value` and `extra` fields. 
3. What other non-event-specific pieces of data need to be attached to an event in its `extra` field, such as a user's fxa uid or random item ids.
4. Why we are collecting these events, and what analyses they will be used for.

The data collection plan will assume usage of [this](https://github.com/mozilla-mobile/telemetry-ios/) library. Where possible, I will give example syntax for event registration and recording.

### PR DESCRIPTION
**Pull request by irrationalagent**
Resolves #130 

OK, here is a first pass at a list of events we'd like to record given the current functionality (as I understand it anyway). In metrics.md I add a few notes of things I'm unsure about; here's a brief tl;dr:

1. We need to have the fxa uid present in the `extras` field of every event where possible (i.e. after authentication) - we currently do this for the extension but if its more difficult here, let me know.

2. I'm unsure of how hard it is to capture errors in a succinct way if fxa authentication fails for some reason. It would be nice to have a short, interpretable error code or message that is added to the extras field on the events that tracks authentication failure / success. 

3. I went ahead and defined the required event methods and objects in `TelemetryIntegration.swift` . If I fouled this up let me know.

Let me know how else I can help or if something is not clear or stupid. I think I get the basics of the mechanics here, so willing to try my hand at placing some of the events in myself, if needed.

Thanks!

### CODE REVIEW

**Comment by devinreams** [comment]
For waffle/GitHub organization purposes I just edited the original description to include "Resolves #130" ☝️

File: lockbox-ios/TelemetryIntegration.swift [hunk 1]
```diff
@@ -9,14 +9,29 @@ class TelemetryEventCategory {
...
     public static let background = "background"
     public static let foreground = "foreground"
-    public static let click = "click"
+    public static let tap = "tap"
+    public static let signin = "signin"
 }
 
 class TelemetryEventObject {
     public static let app = "app"
     public static let initStoreButton = "init_store_button"
```
Comment (sashei) [inline lockbox-ios/TelemetryIntegration.swift]: These three lines could be removed as part of this.

File: lockbox-ios/TelemetryIntegration.swift [hunk 2]
```diff
@@ -9,14 +9,29 @@ class TelemetryEventCategory {
 }
 
 class TelemetryEventMethod {
+    public static let startup = "startup"
     public static let background = "background"
```
Comment (sashei) [inline lockbox-ios/TelemetryIntegration.swift]: I don't see a "backgrounded" or "foregrounded" event in the above enumeration, so these can be removed as well.

File: docs/metrics.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		* Note: If there is an authentication error, let's put it in the extra field here, if possible.
+
+4. When a user taps an item in the entry list:
+	* `category`: action
+	* `method`: tap
+	* `object`: entryList
+	* `value`: nil
+	* `extras`: ["fxauid" : uid, "itemid" : itemid]
+
+5. When a user taps one of the buttons that are shown in the menu displayed after selecting an entry:
```
Comment (sashei) [inline docs/metrics.md]: I think that menu is no longer a part of the iOS designs... can @changecourse confirm?

Comment (changecourse) [reply docs/metrics.md]: Correct... we are removing all functionality around quick-access pane for now, and instead supporting standard chevron, click anywhere on the bar, to access entry details.

**Comment by irrationalagent** [comment]
Thanks for the review, Sasha. I decided to keep the fore/background events after all, as they might be helpful in establishing some general measure of how much time people spend in the app. Not sure what we'd need that for now, but best to have it anyway. I added the spec for it to the md file

**Comment by sashei** [comment]
r+

**Pull request merged**